base/linalg/dense.jl

```diff
@@ -554,37 +554,33 @@ julia> logm(A)
 function logm(A::StridedMatrix{T}) where T
     # If possible, use diagonalization
     if issymmetric(A) && T <: Real
-        return full(logm(Symmetric(A)))
+        return logm(Symmetric(A))
     end
     if ishermitian(A)
-        return full(logm(Hermitian(A)))
+        return logm(Hermitian(A))
     end
 
     # Use Schur decomposition
     n = checksquare(A)
     if istriu(A)
         retmat = full(logm(UpperTriangular(complex(A))))
```

Comment:
purely a style point, but looks like we don't need a variable for retmat with this version, could replace each `retmat =` with `return` ?